src/gpuarray_buffer_cuda.c

```diff
@@ -910,6 +973,36 @@ static gpukernel *cuda_newkernel(void *c, unsigned int count,
           cuda_exit(ctx);
           return NULL;
         }
+        if (compile_cache == NULL)
```

Comment:
It's ok.  The check above is only to see if something is in the cache.  So if there is no cache, nothing's in there.